tests/model_registry/rbac/conftest.py

```diff
@@ -162,11 +198,39 @@ def mr_access_role_binding(
         subjects_name=f"system:serviceaccounts:{sa_namespace.name}",
         subjects_api_group="rbac.authorization.k8s.io",  # This is the default apiGroup for Group kind
         # Role reference parameters
-        role_ref_kind="Role",
+        role_ref_kind=mr_access_role.kind,
         role_ref_name=mr_access_role.name,
         label=binding_labels,
         wait_for_resource=True,
     ) as binding:
         LOGGER.info(f"RoleBinding {binding.name} created successfully.")
         yield binding
         LOGGER.info(f"RoleBinding {binding.name} deletion initiated by context manager.")
+
+
+@pytest.fixture(scope="function")
+def user_in_group_context() -> Callable[[str], ContextManager[None]]:
+    """
+    Fixture to add and remove a user from the model-registry-users group.
+    Returns a context manager that adds the user on entry and removes on exit.
+    """
+
+    @contextmanager
+    def _context(user_name: str) -> Generator[None, None, None]:
+        run_command(command=["oc", "adm", "groups", "add-users", "model-registry-users", user_name])
+        max_attempts = 5
+        for attempt in range(max_attempts):
+            result = run_command(command=["oc", "get", "group", "model-registry-users", "-o", "jsonpath='{.users}'"])
+            if user_name in result[1]:
+                LOGGER.info(f"User {user_name} added to model-registry-users group successfully.")
+                break
+            if attempt <= max_attempts - 1:
+                time.sleep(SLEEP_TIME)  # noqa: FCN001
+        else:
+            pytest.fail(f"User {user_name} not added to model-registry-users group after {max_attempts} attempts.")
+        try:
+            yield
+        finally:
```

Comment:
You don't need finally. If the command failed to add, then we need to remove it.